modules/channel-twilio/src/backend/client.ts

```diff
@@ -88,101 +108,50 @@ export class TwilioClient {
+    const { value } = option
     return {
-      type,
-      text: type === 'say_something' ? value : label,
-      payload: value
+      type: 'text',
+      text: value
     }
   }
 
-  async handleOutgoingEvent(event: sdk.IO.OutgoingEvent, next: sdk.IO.MiddlewareNextCallback) {
-    const payload = event.payload
-
-    if (payload.quick_replies) {
-      await this.sendChoices(event, payload.quick_replies)
-    } else if (payload.options) {
-      await this.sendOptions(
-        event,
-        event.payload.message,
-        {},
-        event.payload.options.map(x => ({ ...x, type: 'quick_reply' }))
-      )
-    } else if (payload.type === 'text') {
-      await this.sendMessage(event, {
-        body: payload.text
-      })
-    } else if (payload.type === 'file') {
-      await this.sendMessage(event, {
-        body: payload.title,
-        mediaUrl: payload.url
-      })
-    } else if (payload.type === 'carousel') {
-      await this.sendCarousel(event, payload)
-    }
-
-    await this.bp.experimental.messages
-      .forBot(this.botId)
-      .create(event.threadId, payload, undefined, event.id, event.incomingEventId)
-
-    next(undefined, false)
-  }
-
-  async sendChoices(event: sdk.IO.Event, choices: ChoiceOption[]) {
-    const options: MessageOption[] = choices.map(x => ({
-      label: x.title,
-      value: x.payload,
-      type: 'quick_reply'
-    }))
-    await this.sendOptions(event, event.payload.text, {}, options)
+  async prepareIndexResponse(event: sdk.IO.OutgoingEvent, options: MessageOption[]) {
+    await this.kvs.set(this.getKvsKey(event.target, event.threadId), options, undefined, '10m')
   }
 
-  async sendCarousel(event: sdk.IO.Event, payload: any) {
-    for (const { subtitle, title, picture, buttons } of payload.elements) {
-      const body = `${title}\n\n${subtitle ? subtitle : ''}`
-
-      const options: MessageOption[] = []
-      for (const button of buttons || []) {
-        const title = button.title as string
-
-        if (button.type === 'open_url') {
-          options.push({ label: `${title} : ${button.url}`, value: undefined, type: 'url' })
-        } else if (button.type === 'postback') {
-          options.push({ label: title, value: button.payload, type: 'postback' })
-        } else if (button.type === 'say_something') {
-          options.push({ label: title, value: button.text as string, type: 'say_something' })
-        }
-      }
+  async handleOutgoingEvent(event: sdk.IO.OutgoingEvent, next: sdk.IO.MiddlewareNextCallback) {
+    const botPhoneNumber = await this.bp.kvs.forBot(this.botId).get(`twilio-number-${event.threadId}`)
 
-      const args = { mediaUrl: picture ? picture : undefined }
-      await this.sendOptions(event, body, args, options)
+    const context: TwilioContext = {
+      bp: this.bp,
+      event,
+      client: this.twilio,
+      handlers: [],
+      payload: _.cloneDeep(event.payload),
+      botUrl: process.EXTERNAL_URL,
+      messages: [],
```

Comment:
First time I notice this messages array, what is it for ?

Reply:
This the array of messages that the senders use to send things. Basically a renderer puts a message in there, and then another renderer could modify that message before it is sent by a sender.